kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java

```diff
@@ -913,13 +913,13 @@ public void run() {
                 }
                 connected = true;
             } catch (KuraConnectException | KuraStoreException e) {
-                logger.warn("Connect failed", e);
-
+                logger.warn("DataService failure occured: ", e);
+                
                 if (DataServiceImpl.this.dataServiceOptions.isConnectionRecoveryEnabled()) {
 
                     if (isAuthenticationException(e) || DataServiceImpl.this.connectionAttempts
                             .getAndIncrement() < DataServiceImpl.this.dataServiceOptions
-                                    .getRecoveryMaximumAllowedFailures()) {
+                                    .getRecoveryMaximumAllowedFailures() || e instanceof KuraStoreException) {
```

Comment:
We should move the new check before of the `this.connectionAttempts.getAndIncrement()` call otherwise the remaining attempt counter will be decreased unnecessarily.